reana_server/scheduler.py

```diff
@@ -33,22 +33,39 @@
 from reana_server.config import (
     REANA_SCHEDULER_REQUEUE_SLEEP,
     REANA_SCHEDULER_REQUEUE_COUNT,
+    REANA_WORKFLOW_SCHEDULING_READINESS_CHECK,
 )
 from reana_server.status import NodesStatus
 
 
-def check_memory_availability(workflow_min_job_memory):
+def check_memory_availability(
+    workflow_min_job_memory: Optional[float],
+) -> Tuple[bool, Optional[str]]:
```

Comment:
What do you think about returning an `Optional[str]` instead of a tuple? `None` would mean that there are no errors, while a `str` would represent the error message.

In `reana_ready`:
```python
for check_condition in conditions:
    error = check_condition()
    if error is not None:
        return error
```

Reply:
Thanks for a nice suggestion